### DISCUSSION TITLE
What is the best practice to manage packages in termux?

### DISCUSSION THREAD

**Discussion by Freed-Wu** [termux/termux-packages#19388] [Q&A]
Currently, termux-packages only host PYPI/npm/... packages which contain binary files like dynamic linked library, or some pure script package depended by a package which contain binary files. So it looks like user should mixed use apt (pkg)/pip/npm/...

So what is the best practice to manage packages in termux? I thought:

1. If a package is available in termux's apt (pkg), `apt install` it.
2. else, `pip/npm/...` install it.

However, if a package is not available in termux's apt (pkg), its dependence is available in termux's apt (pkg), what should we do?

Method 1:

1. search its all dependencies recursively to find which packages are available in termux's apt (pkg), `apt install` them.
2. Then `pip/npm/...` install we required package. `pip/npm/...` will skip those installed dependencies by apt (pkg)

Method 2:

1. Just `pip/npm/...` install we required packages and let these binary files are built from source.

The troublesome of method 1:

1. search which dependencies are available in termux is a trouble ting.
2. when we uninstall the package the dependencies installed by apt (pkg) cannot be uninstalled by `pip/npm`
because those dependencies installed by apt (pkg) are installed explicitly.

Method 2:

1. Build from source is slow and impossible in some low performance device.
2. sometime it will be conflict due to same position of two package managers.

Which method is official recommended method?

Or is there third method? (repackage PYPI/npm packages to termux package and host it in user's apt source to make all dependencies are available in apt (pkg) then `apt install`)?

I just be comfused. Thanks for any answer!

**Comment by HusniMuhammad** [comment]
Use proot-distro debian within termux
